### ISSUE TITLE
How to use hook methods getState, setState, and subscribe with the react context hook?

### ISSUE THREAD

**Issue by omasback** [pmndrs/zustand#377]
Trying out the new context hook. Works great for the base case of selectors, but there doesn't seem to be a way to use the methods `useStore.getState`, `useStore.setState`, and especially `useStore.subscribe`. I can probably work around the lack of get and set, but `subscribe` is important for doing renderless updates with react-spring.

Looking at the `zustand/context` code, it's not obvious to me how this could be implemented. But maybe somebody smarter than me has an idea?

**Comment by dai-shi** [comment]
Ah, nice catch. I haven't thought about the use case, but yeah `subscribe` seems important.
IMO, `useStore.subscribe` is not very well designed for context usage. So, instead of extending `useStore`, I'd suggest adding `useStoreApi`.

```js
const { Provider, useStore, useStoreApi } = createContext()
```

Would you or anyone like to work?

**Comment by dai-shi** [comment]
Fixed by #380. Thanks for your work!
